datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -243,26 +243,36 @@ def _job_loop(self):
                     break
                 self._run_job_rate_limited()
         except self._expected_db_exceptions as e:
-            self._log.warning(
-                "[%s] Job loop database error: %s",
-                self._job_tags_str,
-                e,
-                exc_info=self._log.getEffectiveLevel() == logging.DEBUG,
-            )
-            self._check.count(
-                "dd.{}.async_job.error".format(self._dbms),
-                1,
-                tags=self._job_tags + ["error:database-{}".format(type(e))],
-                raw=True,
-            )
+            # canceling can cause exceptions if the connection is closed the middle of the check run
+            # in this case we still want to report it as a cancellation instead of a crash
+            if self._cancel_event.isSet():
+                self._log.info("[%s] Job loop cancelled", self._job_tags_str)
+                self._check.count("dd.{}.async_job.cancel".format(self._dbms), 1, tags=self._job_tags, raw=True)
+            else:
+                self._log.warning(
+                    "[%s] Job loop database error: %s",
+                    self._job_tags_str,
+                    e,
+                    exc_info=self._log.getEffectiveLevel() == logging.DEBUG,
+                )
+                self._check.count(
+                    "dd.{}.async_job.error".format(self._dbms),
+                    1,
+                    tags=self._job_tags + ["error:database-{}".format(type(e))],
+                    raw=True,
+                )
         except Exception as e:
-            self._log.exception("[%s] Job loop crash", self._job_tags_str)
-            self._check.count(
-                "dd.{}.async_job.error".format(self._dbms),
-                1,
-                tags=self._job_tags + ["error:crash-{}".format(type(e))],
-                raw=True,
-            )
+            if self._cancel_event.isSet():
+                self._log.info("[%s] Job loop cancelled", self._job_tags_str)
```

Comment:
nit so debugging is easier- can we update this log line to be different from https://github.com/DataDog/integrations-core/pull/10934/files#diff-a978df1de9080327a98304d0e39e0dbd33e9353e7b51e62f2d0b494a10c60dc0R249?

Reply:
I refactored to simplify and now that log happens only in one place. Could you take another look? 
https://github.com/DataDog/integrations-core/pull/10934/commits/f5d1128e8ffd7d2430e357c9df5cf0d26290b463

Reply:
Looks good! The refactor makes more sense!